src/main/java/gndata/lib/util/OntologyHelper.java

```diff
@@ -0,0 +1,103 @@
+package gndata.lib.util;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import com.hp.hpl.jena.ontology.*;
+import com.hp.hpl.jena.rdf.model.Resource;
+import com.hp.hpl.jena.vocabulary.RDF;
+
+/**
+ * Created by andrey on 11.05.15.
+ */
+public class OntologyHelper {
+
+    private OntModel ontology;
+
+    public OntologyHelper(OntModel ontology) {
+        this.ontology = ontology;
+    }
+
+    /* ontology Class methods */
+
+    public List<OntClass> listClasses() {
+        return ontology.listClasses().toList();
+    }
+
+    public List<OntClass> listClasses(Resource res) {
+        List<String> types = res.listProperties(RDF.type).toList().stream()
+                .map(st -> st.getObject().asResource().getURI())
+                .collect(Collectors.toList());
+
+        return ontology.listClasses().toList().stream()
+                .filter(cls -> types.contains(cls.getURI()))
+                .collect(Collectors.toList());
+    }
+
+    public List<OntClass> listRelated(OntClass cls) {
+        List<OntClass> range = new ArrayList<>();
+
+        cls.listDeclaredProperties().toList().stream()
+                .filter(OntProperty::isObjectProperty)
+                .forEach(prop -> {
+                    prop.listRange().toList().forEach(ontRes -> {
```

Comment:
prop.listRange() does not yield anything either, even when the OntProperty filter above is removed.